src/couch_db_updater.erl

```diff
@@ -43,9 +43,10 @@ init({DbName, Filepath, Fd, Options}) ->
         ok = couch_file:write_header(Fd, Header),
         % delete any old compaction files that might be hanging around
         RootDir = config:get("couchdb", "database_dir", "."),
-        couch_file:delete(RootDir, Filepath ++ ".compact"),
-        couch_file:delete(RootDir, Filepath ++ ".compact.data"),
-        couch_file:delete(RootDir, Filepath ++ ".compact.meta");
+        DelOpts = [compaction],
+        couch_server:delete_file(RootDir, Filepath ++ ".compact", DelOpts),
+        couch_server:delete_file(RootDir, Filepath ++ ".compact.data",DelOpts),
```

Comment:
We code use the same approach as [here](https://github.com/apache/couchdb-couch/blob/master/src/couch_server.erl#L457:L459). We could also extract this common code into helper function.
While doing so I would also remove [this line](https://github.com/apache/couchdb-couch/blob/master/src/couch_server.erl#L460). Since the very same file is removed couple of lines above.